packages/execution-environments/src/common/endowments/index.ts

```diff
@@ -75,10 +80,20 @@ export function createEndowments(
         // exist in our current environment.
         throw new Error(`Unknown endowment: "${endowmentName}".`);
       }
-      return allEndowments;
+      return { allEndowments, teardowns };
+    },
+    {
+      allEndowments: { wallet } as Record<string, unknown>,
+      teardowns: [] as (() => void)[],
     },
-    { wallet } as Record<string, unknown>,
   );
+
+  const teardown = () => {
+    for (const f of result.teardowns) {
+      f();
+    }
+  };
```

Comment:
A more functional alternative with a better variable name:
```suggestion
  const teardown = () => result.teardowns.forEach(
    (teardownFunction) => teardownFunction(),
  );
```

Reply:
Done!